src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -122,6 +172,50 @@ public LoweringVisitor(DocumentIntermediateNode document, IntermediateNodeBuilde
 
             public string FilePath { get; set; }
 
+            public override void VisitDirectiveToken(DirectiveTokenChunkGenerator chunkGenerator, Span span)
```

Comment:
These visit methods are a direct copy from the `MainSourceVisitor` (this class is used for visiting the primary document and the import documents). There was no need to have a separate import pass anymore because we need to remove overridden directives; to do that I added all directives to the document and then decisively remove them / log diagnostics before we finish the phase.